src/core/js/router.js

```diff
@@ -103,37 +103,51 @@ define([
         handleId: function(id) {
 
             var currentModel = Adapt.findById(id);
+            var type = '';
+            
+            if (!currentModel) {
+                return;
+            } else {
+                type = currentModel.get('_type');
+            }
 
-            switch (currentModel.get('_type')) {
-                case 'page': case 'menu':
-                    this.showLoading();
-                    this.removeViews();
-
-                    this.setContentObjectToVisited(currentModel);
-
-                    if (currentModel.get('_type') == 'page') {
-                        var location = 'page-' + id;
-                        this.updateLocation(location, 'page', id);
-                        Adapt.once('pageView:ready', function() {
-                            //allow navigation
-                            Adapt.router.set('_canNavigate', true, {pluginName: "adapt"});
-                        });
-                        Adapt.trigger('router:page', currentModel);
-                        this.$wrapper.append(new PageView({model:currentModel}).$el);
+            switch (type) {
+                case 'page':
+                case 'menu':
+                    if (currentModel.get('_isLocked') && Adapt.config.get('_forceRouteLocking')) {
+                        console.log('Unable to navigate to locked id: ' + id);
+                        Adapt.router.set('_canNavigate', true, {pluginName: "adapt"});
+                        return Backbone.history.history.back();
```

Comment:
How about also checking for availability of previous route and if its not been there then navigate to the course?
```javascript
if(Adapt.location._previousId === undefined) {
    return this.navigate("#/", {trigger:true, replace:true});
} else {
    return Backbone.history.history.back();
}
```